src/DIRAC/Resources/IdProvider/OidcAgentIdProvider.py

```diff
@@ -0,0 +1,36 @@
+"""
+IdProvider using oidc-agent set up in the running environment
+"""
+
+import subprocess
+
+from DIRAC import S_OK, S_ERROR
+from DIRAC.Core.Utilities.Subprocess import shellCall
+from DIRAC.Resources.IdProvider.IdProvider import IdProvider
+from DIRAC.FrameworkSystem.private.authorization.utils.Tokens import OAuth2Token
+
+
+class OidcAgentIdProvider(IdProvider):
```

Comment:
I am probably missing some details here, but it seems like there is an "architectural issue": In the `IdProvider` directory, I see a mix between identity providers (`IAM`, `CheckIn`), protocols (`OAuth2`) and methods to get tokens (`OIDCAgent`).

In the current model, you cannot get IAM/CheckIn scopes from a group if you use `OIDCAgentIdProvider`, can you?
Also, there is no `getToken()` in `OAuth2IdProvider`, is it expected? (`OAuth2IdProvider.fetchToken()` looks very similar though).

Here is a suggestion:


![IdProvidersUML](https://user-images.githubusercontent.com/19813740/216339175-930e972a-f507-4bdf-8a27-afa79bc7c726.jpg)

Reply:
I second @aldbr here. I would just integrate also the suggestion of @chrisburr above to make `IdProvider`, and also this new (?) `AccessStrategy` modules Abstract Classes (there would no be more "extensions" rather than "implementations" then...)

Q to @aldbr : in the drawing above are "OAuth2" and "@OAuth2lib" the "same" (apart from they inherit from)?

And in general, I think we agreed that we might not really want to support oidc-agent.

Reply:
By the way, I wonder whether the term of `IdProvider` is adapted in this context.
In the papers I have read so far, `IAM` and `CheckIn` are defined as `Token Issuer` or `OpenID Providers`, while `IdProvider` refers to services that actually store information about the users such as `Google`, `Facebook`, `CERN?`.

What do you think?

Reply:
> Q to @aldbr : in the drawing above are "OAuth2" and "@OAuth2lib" the "same" (apart from they inherit from)?

Yes I didn't specify it sorry, but `OAuth2` in the current model = `OAuth2lib` in the suggested model.

Reply:
The OidcAgentIdProvider removed in favor of a method in the OAuth2IdProvider class